tests/foreman/cli/test_product.py

```diff
@@ -246,3 +247,101 @@ def test_product_list_with_default_settings(self):
+        })
+        http_proxy_b = HttpProxy.create({
+            'name': gen_string('alpha', 15),
+            'url': http_proxy_url_b,
+            'organization-id': self.org['id'],
+        })
+        proxy_fqdn = http_proxy_b['url'].split(":")[1].strip("//")
+        # Create products and repositories
+        product_a = make_product({
+            u'organization-id': self.org['id'],
+        })
+        product_b = make_product({
+            u'organization-id': self.org['id'],
+        })
+        repo_a1 = make_repository({
+            'product-id': product_a['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'none'
+        })
+        repo_a2 = make_repository({
+            'product-id': product_a['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'use_selected_http_proxy',
+            'http-proxy-id': http_proxy_a['id']
+        })
+        repo_b1 = make_repository({
+            'product-id': product_b['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'none'
+        })
+        repo_b2 = make_repository({
+            'product-id': product_b['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'global_default_http_proxy',
+        })
+        # Add http_proxy to products
+        Product.update_proxy({
+            'ids': '{},{}'.format(product_a['id'], product_b['id']),
+            'http-proxy-policy': 'use_selected_http_proxy',
+            'http-proxy-id': http_proxy_b['id']
+        })
+        repo_a1 = Repository.info({'id': repo_a1['id']})
+        repo_a2 = Repository.info({'id': repo_a2['id']})
+        repo_b1 = Repository.info({'id': repo_b1['id']})
+        repo_b2 = Repository.info({'id': repo_b2['id']})
+        # Update following assert statements when BZ#1777713 is fixed.
+        assert repo_a1['product']['http-proxy-policy'] == "use_selected_http_proxy"
+        assert repo_a2['product']['http-proxy-policy'] == "use_selected_http_proxy"
+        assert repo_b1['product']['http-proxy-policy'] == "use_selected_http_proxy"
+        assert repo_b2['product']['http-proxy-policy'] == "use_selected_http_proxy"
+        assert repo_a1['http-proxy']['id'] == http_proxy_b['id']
+        assert repo_a2['http-proxy']['id'] == http_proxy_b['id']
+        assert repo_b1['http-proxy']['id'] == http_proxy_b['id']
+        assert repo_b2['http-proxy']['id'] == http_proxy_b['id']
+        # check if proxy fqdn is present in log during sync
+        Product.synchronize({
+            'id': product_a['id'],
+            'organization-id': self.org['id'],
+            'async': True
+        })
+        result = ssh.command('grep -F {} /var/log/messages'.format(proxy_fqdn))
+        assert result.return_code == 0
+        # Add http_proxy to products
+        Product.update_proxy({
+            'ids': '{},{}'.format(product_a['id'], product_b['id']),
+            'http-proxy-policy': 'none',
+        })
```

Comment:
Just a suggestion , If you could add below steps as well after updated the proxy to `none`:
1. Try a product sync 
2. see `/var/log/messages` to check no proxy using now  during sync.

Reply:
@vijay8451 updated the PR.